receiver/scraper.go

```diff
@@ -0,0 +1,120 @@
+		cfg: cfg,
+		set: &set,
+		mb:  metadata.NewMetricsBuilder(cfg.MetricsConfig),
+	}
+}
+
+func (e *expVarScraper) start(_ context.Context, host component.Host) error {
+	httpClient, err := e.cfg.HTTPClientSettings.ToClient(host.GetExtensions(), e.set.TelemetrySettings)
+	if err != nil {
+		return err
+	}
+	e.httpClient = httpClient
+	return nil
+}
+
+func (e *expVarScraper) scrape(_ context.Context) (pmetric.Metrics, error) {
+	emptyMetrics := pmetric.NewMetrics()
+	resp, err := e.httpClient.Get(e.cfg.Endpoint)
+	if err != nil {
+		return emptyMetrics, err
+	}
+	defer resp.Body.Close()
+	if resp.StatusCode != http.StatusOK {
+		return emptyMetrics, fmt.Errorf("expected 200 but received %d status code", resp.StatusCode)
+	}
+
+	result, err := decodeResponseBody(resp.Body)
+	if err != nil {
+		return emptyMetrics, err
+	}
+	memStats := result.MemStats
+	if memStats == nil {
+		return emptyMetrics, fmt.Errorf("unmarshalled memstats data is nil")
+	}
+
+	now := pcommon.NewTimestampFromTime(time.Now())
+
+	e.mb.RecordProcessRuntimeMemstatsTotalAllocDataPoint(now, int64(memStats.TotalAlloc))
+	e.mb.RecordProcessRuntimeMemstatsSysDataPoint(now, int64(memStats.Sys))
+	e.mb.RecordProcessRuntimeMemstatsLookupsDataPoint(now, int64(memStats.Lookups))
+	e.mb.RecordProcessRuntimeMemstatsMallocsDataPoint(now, int64(memStats.Mallocs))
+	e.mb.RecordProcessRuntimeMemstatsFreesDataPoint(now, int64(memStats.Frees))
+	e.mb.RecordProcessRuntimeMemstatsHeapAllocDataPoint(now, int64(memStats.HeapAlloc))
+	e.mb.RecordProcessRuntimeMemstatsHeapSysDataPoint(now, int64(memStats.HeapSys))
+	e.mb.RecordProcessRuntimeMemstatsHeapIdleDataPoint(now, int64(memStats.HeapIdle))
+	e.mb.RecordProcessRuntimeMemstatsHeapInuseDataPoint(now, int64(memStats.HeapInuse))
+	e.mb.RecordProcessRuntimeMemstatsHeapReleasedDataPoint(now, int64(memStats.HeapReleased))
+	e.mb.RecordProcessRuntimeMemstatsHeapObjectsDataPoint(now, int64(memStats.HeapObjects))
+	e.mb.RecordProcessRuntimeMemstatsStackInuseDataPoint(now, int64(memStats.StackInuse))
+	e.mb.RecordProcessRuntimeMemstatsStackSysDataPoint(now, int64(memStats.StackSys))
+	e.mb.RecordProcessRuntimeMemstatsMspanInuseDataPoint(now, int64(memStats.MSpanInuse))
+	e.mb.RecordProcessRuntimeMemstatsMspanSysDataPoint(now, int64(memStats.MSpanSys))
+	e.mb.RecordProcessRuntimeMemstatsMcacheInuseDataPoint(now, int64(memStats.MCacheInuse))
+	e.mb.RecordProcessRuntimeMemstatsMspanSysDataPoint(now, int64(memStats.MCacheSys))
+	e.mb.RecordProcessRuntimeMemstatsBuckHashSysDataPoint(now, int64(memStats.BuckHashSys))
+	e.mb.RecordProcessRuntimeMemstatsGcSysDataPoint(now, int64(memStats.GCSys))
+	e.mb.RecordProcessRuntimeMemstatsOtherSysDataPoint(now, int64(memStats.OtherSys))
+	e.mb.RecordProcessRuntimeMemstatsNextGcDataPoint(now, int64(memStats.NextGC))
+	e.mb.RecordProcessRuntimeMemstatsPauseTotalDataPoint(now, int64(memStats.PauseTotalNs))
+	e.mb.RecordProcessRuntimeMemstatsLastPauseDataPoint(now, int64(memStats.PauseNs[(memStats.NumGC+255)%256]))
```

Comment:
How come we are doing some modulo math here?

Reply:
From the memstats docs: 

```
// PauseNs is a circular buffer of recent GC stop-the-world
// pause times in nanoseconds.
//
// The most recent pause is at PauseNs[(NumGC+255)%256]. In
// general, PauseNs[N%256] records the time paused in the most
// recent N%256th GC cycle. There may be multiple pauses per
// GC cycle; this is the sum of all pauses during a cycle.
```

I'm trying to get the most recent pause time. Do you think there should be a comment here?

Reply:
A comment would make sense here to avoid any confusion of where this came about.

Reply:
Done